Source/AdUnits/Native/NativeAsset.swift

```diff
@@ -0,0 +1,126 @@
+/*   Copyright 2018-2019 Prebid.org, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import UIKit
+
+public class NativeAsset: NSObject {
+    
+    var id: NSInteger!
+    public var required: Bool = false
+    
+    internal init(isRequired: Bool) {
+        super.init()
+        required = isRequired
+    }
+    
+    func getAssetObject() -> [AnyHashable: Any] {
+        var asset: [AnyHashable: Any] = [:]
+        
+        //asset["id"] = Int.random(in: 0...1000)
+        asset["required"] = Int(truncating: NSNumber(value: required))
+        
+        if let titleObject = self as? NativeAssetTitle {
+            asset["title"] = titleObject.getTitleObject()
+        } else  if let imageObject = self as? NativeAssetImage {
+            asset["img"] = imageObject.getImageObject()
+        } else if let dataObject = self as? NativeAssetData {
+            asset["data"] = dataObject.getDataObject()
+        }
+        return asset
+    }
+    
+}
+
+@objcMembers public class NativeAssetTitle: NativeAsset {
+    
+    var length: NSInteger! = 25
+    
+    public var ext: AnyObject?
+    
+    public required init(length: NSInteger, required: Bool) {
+        super.init(isRequired: required)
+        self.length = length
+    }
+    
+    func getTitleObject() -> [AnyHashable: Any] {
+        var title: [AnyHashable: Any] = [:]
+        
+        title["len"] = length
+        title["ext"] = ext
+        
+        return title
+    }
+    
+}
+
+public class NativeAssetImage: NativeAsset {
+    
+    public var type: ImageAsset?
+    public var width: Int?
+    public var widthMin: Int?
+    public var height: Int?
+    public var heightMin: Int?
+    public var mimes: Array<String>?
+    public var ext: AnyObject?
+    
+    public required init(minimumWidth: Int, minimumHeight: Int, required: Bool) {
+        super.init(isRequired: required)
```

Comment:
What if the publisher wants just exact width and height, but not mW and mH?
Can you provide an alternative Init method?